xds/src/main/java/io/grpc/xds/XdsLoadBalancerProvider.java

```diff
@@ -73,8 +74,7 @@ public ConfigOrError parseLoadBalancingPolicyConfig(
   static ConfigOrError parseLoadBalancingConfigPolicy(
       Map<String, ?> rawLoadBalancingPolicyConfig, LoadBalancerRegistry registry) {
     try {
-      LbConfig newLbConfig =
-          ServiceConfigUtil.unwrapLoadBalancingConfig(rawLoadBalancingPolicyConfig);
+      LbConfig newLbConfig = new LbConfig(XDS_POLICY_NAME, rawLoadBalancingPolicyConfig);
```

Comment:
`LbConfig.policyName` is never read from any of the following methods. Why wrap it by adding something that's never read?

Reply:
Because the API `ServiceConfigUtil.getBalancerNameFromXdsConfig()` used below is asking for a `LbConfig`, this maybe need further refactoring, but the CL only fixes the bug.

Reply:
I can add `checkArgument` in `getBalancerNameFromXdsConfig()` etc to use `policyName` is you like :)

Reply:
Refactored.